pandas/core/sparse/frame.py

```diff
@@ -547,6 +548,31 @@ def xs(self, key, axis=0, copy=False):
     # ----------------------------------------------------------------------
     # Arithmetic-related methods
 
+    def _wrap_dispatched_op(self, result, other, func, axis=None):
+        """
+        Wrap the result of an arithmetic/comparison operation performed
+        via ops.dispatch_to_series in a properly-indexed DataFrame.
+
+        Parameters
+        ----------
+        result : dict[int:Series]
+        other : DataFrame, Series, scalar
+        func : binary function
+        axis : {None, "columns"}
+
+        Returns
+        -------
+        SparseDataFrame
+        """
+
+        fill_value = self._get_op_result_fill_value(other, func, axis)
+
+        res = self._constructor(result, index=self.index,
+                                columns=self.columns,
+                                default_fill_value=fill_value)
+
+        return res.__finalize__(self)
```

Comment:
why would this call finalize? why would the DataFrame one not?

Reply:
No idea, that's why I opened #23028.

Reply:
I'll do this, but am not wild about it.  Current `__finalize__/_metadata` handling is quarter-assed, and I'd prefer to leave it that way until it is specifically addressed than make it half-assed.  Still though, will update.

Reply:
Hmm this actually makes things _less_ consistent within DataFrame (though slightly more consistent between DataFrame - SparseDataFrame) since lots of ops _dont_ go through this path.

Reply:
can you just call the super function here?